### PR TITLE
Support anonymous binding better

### PR DESCRIPTION
**Pull request by monkeyWzr**
Support anonymous binding better. Skip unnecessary updates and make `removeEventListener` work.

[https://github.com/ndelvalle/v-click-outside/issues/34](https://github.com/ndelvalle/v-click-outside/issues/34)

### CODE REVIEW

File: lib/v-click-outside.js [hunk 1]
```diff
@@ -71,19 +71,23 @@ function bind(el, { value }) {
...
-function update(el, { value }) {
+function update(el, { value, oldValue }) {
   const { events, handler, middleware, isActive } = processDirectiveArguments(value)
 
   if (!isActive) {
     removeInstance(el)
     return
   }
 
+  if (JSON.stringify(value) === JSON.stringify(oldValue)) {
```
Comment (ndelvalle) [inline lib/v-click-outside.js]: Should we move this to be the first thing to be executed in the `update` method? Above `isActive` check and `processDirectiveArguments`?

**Comment by ndelvalle** [comment]
Thanks for the contribution @monkeyWzr, I just had a small comment, let me know what you think 🙂
This PR also solves #62, what do you think @trongthanh?

**Comment by trongthanh** [comment]
It looks fine. Please publish to alpha tag so that I can help test it.

File: lib/v-click-outside.js (hunk 1 above)
Comment (monkeyWzr) [reply lib/v-click-outside.js]: I think you are right. I modified that. Thank you!

**Comment by ndelvalle** [comment]
@monkeyWzr @trongthanh  v-click-outside 2.1.2-0 published 🎉. Thanks for the contribution, let me know how the test goes @trongthanh :)

**Pull request merged**